src/realm/sync/client.cpp

```diff
@@ -780,44 +786,44 @@ void SessionImpl::process_pending_flx_bootstrap()
     VersionInfo new_version;
     DownloadCursor download_cursor;
     int64_t query_version = -1;
-    try {
-        while (bootstrap_store->has_pending()) {
-            auto pending_batch = bootstrap_store->peek_pending(batch_size_in_bytes);
-            if (!pending_batch.progress) {
-                logger.info("Incomplete pending bootstrap found for query version %1", pending_batch.query_version);
-                bootstrap_store->clear();
-                return;
-            }
+    while (bootstrap_store->has_pending()) {
+        auto pending_batch = bootstrap_store->peek_pending(batch_size_in_bytes);
+        if (!pending_batch.progress) {
+            logger.info("Incomplete pending bootstrap found for query version %1", pending_batch.query_version);
+            bootstrap_store->clear();
+            return;
+        }
 
-            auto batch_state =
-                pending_batch.remaining > 0 ? DownloadBatchState::MoreToCome : DownloadBatchState::LastInBatch;
-            uint64_t downloadable_bytes = 0;
-            query_version = pending_batch.query_version;
+        auto batch_state =
+            pending_batch.remaining > 0 ? DownloadBatchState::MoreToCome : DownloadBatchState::LastInBatch;
+        uint64_t downloadable_bytes = 0;
+        query_version = pending_batch.query_version;
+        bool simulate_integration_error =
```

Comment:
And we thread through this "simulate_integration_error" flag here so we can actually test BadChangeset errors during bootstrap.